numpy/lib/twodim_base.py

```diff
@@ -1013,3 +1042,8 @@ def triu_indices_from(arr, k=0):
     if arr.ndim != 2:
         raise ValueError("input array must be 2-d")
     return triu_indices(arr.shape[-2], k=k, m=arr.shape[-1])
+
+
+# Add documentation for ``like=`` keyword argument
+set_array_function_like_doc(eye)
+set_array_function_like_doc(tri)
```

Comment:
Can't we decorate the function now?

Reply:
That's a good idea, I did try it now but wasn't successful. I tried something like the following:

```python
def set_array_function_like_doc(public_api):
    @functools.wraps(public_api)
    def decorator(*args, **kwargs):
        return public_api(*args, **kwargs)
    if decorator.__doc__ is not None:
        decorator.__doc__ = decorator.__doc__.replace(
            "${ARRAY_FUNCTION_LIKE}",
            array_function_like_doc,
        )
    return decorator
```

The usage of `@functools.wraps(public_api)` should retain the original signature, but it fails in https://github.com/numpy/numpy/blob/64619c55896dad1927afe78c0702a58abf0d038f/numpy/core/overrides.py#L77-L86 as it seems `getargspec` checks for `__code__` which apparently doesn't check for signature in the same way. If there's a different way of doing it that I don't know of let me know, but it seems the other alternative would be to maybe have `getargspec` working in a different way, but I think that would be too intrusive.

Reply:
Can you try [`functools.update_wrapper`](https://docs.python.org/3.8/library/functools.html#functools.update_wrapper)?

Reply:
Thanks @hameerabbasi , same result unfortunately. It seems that `@functools.wraps` is actually a convenience function to `functools.update_wrapper` as per [its documentation](https://docs.python.org/3.8/library/functools.html#functools.wraps).

Reply:
You can use `set_array_function_like_doc` as a decorator easily if you just add `return func` at the end of it.